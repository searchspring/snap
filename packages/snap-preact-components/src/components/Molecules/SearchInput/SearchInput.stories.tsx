import { Fragment, h } from 'preact';

import { ArgsTable, PRIMARY_STORY } from '@storybook/addon-docs/blocks';

import { SearchInput, SearchInputProps } from './SearchInput';
import { Icon, iconPaths } from '../../Atoms/Icon';
import { componentArgs } from '../../../utilities';
import Readme from './readme.md';

export default {
	title: `Molecules/SearchInput`,
	component: SearchInput,
	parameters: {
		docs: {
			page: () => (
				<div>
					<Readme />
					<ArgsTable story={PRIMARY_STORY} />
				</div>
			),
		},
	},
	decorators: [
		(Story: any) => (
			<div
				style={{
					maxWidth: '900px',
					height: '300px',
				}}
			>
				<Story />
			</div>
		),
	],
	argTypes: {
		onChange: {
			description: 'Onchange Callback',
			table: {
				type: {
					summary: 'function',
				},
			},
			action: 'onChange',
		},
		hideIcon: {
			description: 'Hides icon',
			table: {
				type: {
					summary: 'boolean',
				},
				defaultValue: { summary: false },
			},
			control: { type: 'boolean' },
		},
		placeholder: {
			defaultValue: 'Search',
			description: 'Display placeholder text',
			table: {
				type: {
					summary: 'string',
				},
				defaultValue: { summary: 'Search' },
			},
			control: { type: 'text' },
		},
		...componentArgs,
	},
};
export const Default = (args: SearchInputProps) => <SearchInput {...args} />;
