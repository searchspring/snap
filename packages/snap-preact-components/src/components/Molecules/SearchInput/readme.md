# SearchInput

Renders an input element

## Sub-components
- Icon

## Usage

```jsx
<SearchInput onChange={handleChange} placeholder={"placeholder text"}/>
```

### placeholder
The `placeholder` prop sets the input placeholder text. 

```jsx
<SearchInput placeholder={"placeholder text"} />
```

### onChange
The `onChange` prop is invoked when the input's value has been changed. 

```jsx
<SearchInput onChange={handleChange} />
```

### hideIcon
The `hideIcon` prop determines if the icon should be hidden or not. 

```jsx
<SearchInput hideIcon={true} />
```
