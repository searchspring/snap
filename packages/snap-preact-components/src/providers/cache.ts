export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
