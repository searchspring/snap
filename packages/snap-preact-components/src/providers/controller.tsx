import { jsx } from '@emotion/react';
import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractController } from '@searchspring/snap-controller';

const ControllerContext = createContext<null | AbstractController>(null);

export const ControllerProvider = ({ children, controller }: { children: ComponentChildren; controller: AbstractController }) => {
	return <ControllerContext.Provider value={controller}>{children}</ControllerContext.Provider>;
};

export const useController = () => useContext(ControllerContext);

export const withController = (Component: ComponentType) => (props: any) => {
	return <Component controller={useController()} {...props} />;
};
