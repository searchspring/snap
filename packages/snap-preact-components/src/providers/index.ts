export * from './store';
export * from './theme';
