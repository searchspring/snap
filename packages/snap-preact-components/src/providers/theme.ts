import { ThemeProvider as EmotionTheme } from '@emotion/react';
export { css, useTheme, withTheme } from '@emotion/react';

interface ThemeProviderProps {
	theme: Partial<Theme> | ((outerTheme: Theme) => Theme);
	children?: any;
}
interface ThemeProvider {
	(props: ThemeProviderProps): React.ReactElement;
}
export const ThemeProvider: ThemeProvider = EmotionTheme;

export const defaultTheme: Theme = {
	colors: {
		primary: '#3A23AD',
		secondary: '#00cee1',
		hover: '#f8f6fd',
		text: {
			secondary: '#ffffff',
		},
		message: {
			error: '#cc1212',
			warning: '#ecaa15',
			info: '#4c3ce2',
		},
	},
	components: {},
};
export interface Theme {
	variables?: {
		[key: string]: unknown;
	};
	colors?: {
		primary: string;
		secondary: string;
		hover?: string;
		text?: {
			secondary: string;
		};
		message?: {
			error: string;
			warning: string;
			info: string;
		};
	};
	components?: any;
	namedComponents?: any;
}
