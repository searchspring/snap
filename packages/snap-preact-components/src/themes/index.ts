export * from './pike';
export * from './bocachica';

// import { pike } from './pike';
import { bocachica } from './bocachica';

// used in component tests
export default {
	// pike, // TODO: remove pike theme OR add pike styleScripts
	bocachica,
};
