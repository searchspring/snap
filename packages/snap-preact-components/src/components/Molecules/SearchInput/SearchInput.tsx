/** @jsx jsx */
import { h, Fragment } from 'preact';

import { jsx, css } from '@emotion/react';
import classnames from 'classnames';
import { observer } from 'mobx-react-lite';

import { Icon, IconProps } from '../../Atoms/Icon/Icon';
import { defined } from '../../../utilities';
import { Theme, useTheme, CacheProvider } from '../../../providers';
import { ComponentProps, StylingCSS } from '../../../types';

const CSS = {
	searchInput: ({ theme }: { theme: Theme }) =>
		css({
			display: 'flex',
			alignItems: 'center',
			justifyContent: 'center',
			border: `1px solid ${theme.colors?.primary || '#ccc'}`,

			'& .ss__icon': {
				padding: '5px',
			},

			'& .ss__search-input__input': {
				width: '100%',
				outline: 'none',
				border: '0',
				boxSizing: 'border-box',
			},
		}),
};

export const SearchInput = observer((properties: SearchInputProps): JSX.Element => {
	const globalTheme: Theme = useTheme();
	const theme = { ...globalTheme, ...properties.theme };

	let props: SearchInputProps = {
		// default props
		placeholder: 'Search',
		hideIcon: false,
		// global theme
		...globalTheme?.components?.searchInput,
		//props
		...properties,
		...properties.theme?.components?.searchInput,
	};

	const { placeholder, onChange, hideIcon, disableStyles, style, className } = props;

	const subProps: SearchInputSubProps = {
		icon: {
			// default props
			className: 'ss__search-input__icon',
			// global theme
			...globalTheme?.components?.icon,
			// inherited props
			...defined({
				disableStyles,
			}),
			// component theme overrides
			theme: props.theme,
		},
	};

	const styling: { css?: StylingCSS } = {};
	if (!disableStyles) {
		styling.css = [CSS.searchInput({ theme }), style];
	} else if (style) {
		styling.css = [style];
	}
	return (
		<CacheProvider>
			<div {...styling} className={classnames('ss__search-input', className)}>
				{!hideIcon && <Icon {...subProps.icon} icon="search" />}
				<input
					type="text"
					className="ss__search-input__input"
					onChange={(e) => {
						onChange && onChange(e);
					}}
					placeholder={placeholder}
				/>
			</div>
		</CacheProvider>
	);
});

export interface SearchInputProps extends ComponentProps {
	onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
	placeholder?: string;
	hideIcon?: boolean;
}

interface SearchInputSubProps {
	icon: IconProps;
}
