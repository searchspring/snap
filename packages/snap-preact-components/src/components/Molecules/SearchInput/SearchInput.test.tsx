import { h } from 'preact';
import { render } from '@testing-library/preact';
import userEvent from '@testing-library/user-event';

import { ThemeProvider } from '../../../providers/theme';
import themes from '../../../themes';
import { SearchInput } from './SearchInput';

describe('SearchInput Component', () => {
	Object.keys(themes || {}).forEach((themeName) => {
		it(`uses ${themeName} theme`, () => {
			const theme = themes[themeName as keyof typeof themes];
			const rendered = render(<SearchInput theme={theme} placeholder={'hello world'} />);
			expect(rendered.asFragment()).toMatchSnapshot();
		});
	});

	const theme = {
		components: {
			searchInput: {
				placeholder: 'theme',
			},
		},
	};

	it('renders', () => {
		const rendered = render(<SearchInput />);
		const searchInput = rendered.container.querySelector('.ss__search-input');
		expect(searchInput).toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can update placeholder text', () => {
		const placeholder = 'hello';
		const rendered = render(<SearchInput placeholder={placeholder} />);
		const searchInput: HTMLInputElement = rendered.container.querySelector('.ss__search-input__input')!;
		expect(searchInput).toBeInTheDocument();
		expect(searchInput.placeholder).toBe(placeholder);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can invoke onChange callback', () => {
		const onChangeFn = jest.fn();
		const text = 'hello world';
		const rendered = render(<SearchInput onChange={onChangeFn} />);
		const searchInput: HTMLInputElement = rendered.container.querySelector('.ss__search-input__input')!;
		expect(searchInput).toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();

		userEvent.type(searchInput, text);
		expect(searchInput.value).toBe(text);
		expect(onChangeFn).toHaveBeenCalled();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can hide icon using hideIcon prop', () => {
		const rendered = render(<SearchInput hideIcon={true} />);
		const searchInput = rendered.container.querySelector('.ss__search-input__input');
		expect(searchInput).toBeInTheDocument();
		const icon = rendered.container.querySelector('.ss__icon');
		expect(icon).not.toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can disable styling', () => {
		const rendered = render(<SearchInput />);
		const searchInput = rendered.container.querySelector('.ss__search-input');
		expect(searchInput?.classList.length).toBe(2);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('renders with classname', () => {
		const className = 'classy';
		const rendered = render(<SearchInput className={className} />);
		const searchInput = rendered.container.querySelector('.ss__search-input');

		expect(searchInput).toBeInTheDocument();
		expect(searchInput).toHaveClass(className);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is themeable with ThemeProvider', () => {
		const rendered = render(
			<ThemeProvider theme={theme}>
				<SearchInput />
			</ThemeProvider>
		);

		const searchInput: HTMLInputElement = rendered.container.querySelector('.ss__search-input__input')!;
		expect(searchInput).toBeInTheDocument();

		expect(searchInput?.placeholder).toBe(theme.components.searchInput.placeholder);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is themeable with theme prop', () => {
		const rendered = render(<SearchInput theme={theme} />);
		const searchInput: HTMLInputElement = rendered.container.querySelector('.ss__search-input__input')!;
		expect(searchInput).toBeInTheDocument();

		expect(searchInput.placeholder).toBe(theme.components.searchInput.placeholder);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is themeable with theme prop overrides ThemeProvider', () => {
		const componentTheme = {
			components: {
				searchInput: {
					placeholder: 'theme 2',
				},
			},
		};
		const rendered = render(
			<ThemeProvider theme={theme}>
				<SearchInput theme={componentTheme} />
			</ThemeProvider>
		);
		const searchInput: HTMLInputElement | null = rendered.container.querySelector('.ss__search-input__input');
		expect(searchInput).toBeInTheDocument();

		expect(searchInput?.placeholder).toBe(componentTheme.components.searchInput.placeholder);
		expect(rendered.asFragment()).toMatchSnapshot();
	});
});
