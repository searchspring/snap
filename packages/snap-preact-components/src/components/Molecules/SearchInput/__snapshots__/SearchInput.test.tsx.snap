// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component can disable styling 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component can hide icon using hideIcon prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

<div
    class="ss__search-input emotion-0"
  >
    <input
      class="ss__search-input__input"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component can invoke onChange callback 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component can invoke onChange callback 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component can update placeholder text 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="hello"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="theme"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component is themeable with theme prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="theme"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component is themeable with theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="theme 2"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component renders 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component renders with classname 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #ccc;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input classy emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;

exports[`SearchInput Component uses bocachica theme 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: 1px solid #6d7175;
}

.emotion-0 .ss__icon {
  padding: 5px;
}

.emotion-0 .ss__search-input__input {
  width: 100%;
  outline: none;
  border: 0;
  box-sizing: border-box;
}

.emotion-1 {
  fill: #6d7175;
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__search-input emotion-0"
  >
    <svg
      class="ss__icon ss__icon--search ss__search-input__icon emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M38.769 23.692c0-8.313-6.764-15.077-15.077-15.077s-15.077 6.764-15.077 15.077 6.764 15.077 15.077 15.077 15.077-6.764 15.077-15.077zM56 51.692c0 2.356-1.952 4.308-4.308 4.308-1.144 0-2.255-0.471-3.029-1.279l-11.543-11.51c-3.937 2.726-8.649 4.173-13.428 4.173-13.091 0-23.692-10.601-23.692-23.692s10.601-23.692 23.692-23.692 23.692 10.601 23.692 23.692c0 4.779-1.447 9.49-4.173 13.428l11.543 11.543c0.774 0.774 1.245 1.885 1.245 3.029z"
      />
    </svg>
    <input
      class="ss__search-input__input"
      placeholder="hello world"
      type="text"
    />
  </div>
</DocumentFragment>
`;
